spark-statsd-reporter/src/main/java/org/apache/spark/metrics/sink/statsd/StatsdReporter.java

```diff
@@ -112,66 +88,71 @@ private void reportCounters(SortedMap<String, Counter> counters, DatagramSocket
+            addIfMatches(metrics, MetricAttribute.P50, metricFormatter.buildMetricString(name, MetricAttribute.P50.toString(), snapshot.getMedian(), TIMER));
+            addIfMatches(metrics, MetricAttribute.P75, metricFormatter.buildMetricString(name, MetricAttribute.P75.toString(), snapshot.get75thPercentile(), TIMER));
+            addIfMatches(metrics, MetricAttribute.P95, metricFormatter.buildMetricString(name, MetricAttribute.P95.toString(), snapshot.get95thPercentile(), TIMER));
+            addIfMatches(metrics, MetricAttribute.P98, metricFormatter.buildMetricString(name, MetricAttribute.P98.toString(), snapshot.get98thPercentile(), TIMER));
+            addIfMatches(metrics, MetricAttribute.P99, metricFormatter.buildMetricString(name, MetricAttribute.P99.toString(), snapshot.get99thPercentile(), TIMER));
+            addIfMatches(metrics, MetricAttribute.P999, metricFormatter.buildMetricString(name, MetricAttribute.P999.toString(), snapshot.get999thPercentile(), TIMER));
+            send(socket, metrics.toArray(new String[metrics.size()]));
         });
     }
 
     private void reportMeters(SortedMap<String, Meter> meters, DatagramSocket socket) {
         meters.forEach((name, meter) -> {
-            send(socket,
-                    metricFormatter.buildMetricString(name, "count", meter.getCount(), GAUGE),
-                    metricFormatter.buildMetricString(name, "m1_rate", convertRate(meter.getOneMinuteRate()), TIMER),
-                    metricFormatter.buildMetricString(name, "m5_rate", convertRate(meter.getFiveMinuteRate()), TIMER),
-                    metricFormatter.buildMetricString(name, "m15_rate", convertRate(meter.getFifteenMinuteRate()), TIMER),
-                    metricFormatter.buildMetricString(name, "mean_rate", convertRate(meter.getMeanRate()), TIMER)
-            );
+        	List<String> metrics = new ArrayList<>();
+        	addIfMatches(metrics, MetricAttribute.COUNT, metricFormatter.buildMetricString(name, MetricAttribute.COUNT.toString(), meter.getCount(), GAUGE));
+        	addIfMatches(metrics, MetricAttribute.M1_RATE, metricFormatter.buildMetricString(name, MetricAttribute.M1_RATE.toString(), convertRate(meter.getOneMinuteRate()), TIMER));
+        	addIfMatches(metrics, MetricAttribute.M5_RATE, metricFormatter.buildMetricString(name, MetricAttribute.M5_RATE.toString(), convertRate(meter.getFiveMinuteRate()), TIMER));
+        	addIfMatches(metrics, MetricAttribute.M15_RATE, metricFormatter.buildMetricString(name, MetricAttribute.M15_RATE.toString(), convertRate(meter.getFifteenMinuteRate()), TIMER));
+        	addIfMatches(metrics, MetricAttribute.MEAN_RATE, metricFormatter.buildMetricString(name, MetricAttribute.MEAN_RATE.toString(), convertRate(meter.getMeanRate()), TIMER));
+            send(socket, metrics.toArray(new String[metrics.size()]));
```

Comment:
same re: `metrics.toArray(new String[0])`

Reply:
Changed it accordingly.